web/middleware.py

```
@@ -141,3 +143,33 @@
+
+
+class TimeZoneMiddleware:
+    """
+    Middleware that activates the user's timezone preference for each request.
+    It either uses the timezone stored in the session or defaults to TIME_ZONE from settings.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
```

Ruff (0.8.2)

Missing return type annotation for special method `__call__`

(ANN204)